indico/modules/events/registration/util.py

```diff
@@ -523,33 +565,30 @@ def import_registrations_from_csv(regform, fileobj, skip_moderation=True, notify
     registered_emails = {email for (email,) in query}
     used_emails = set()
     todo = []
-    for row_num, row in enumerate(reader, 1):
-        try:
-            first_name, last_name, affiliation, position, phone, email = [to_unicode(value).strip() for value in row]
-            email = email.lower()
-        except ValueError:
-            raise UserValueError(_('Row {}: malformed CSV data - please check that the number of columns is correct')
-                                 .format(row_num))
-
-        if not email:
-            raise UserValueError(_('Row {}: missing e-mail address').format(row_num))
-        if not is_valid_mail(email, multi=False):
-            raise UserValueError(_('Row {}: invalid e-mail address').format(row_num))
-        if not first_name or not last_name:
-            raise UserValueError(_('Row {}: missing first or last name').format(row_num))
-        if email in registered_emails:
-            raise UserValueError(_('Row {}: a registration with this email already exists').format(row_num))
-        if email in used_emails:
-            raise UserValueError(_('Row {}: email address is not unique').format(row_num))
-
-        used_emails.add(email)
-        todo.append({
-            'email': email,
-            'first_name': first_name.title(),
-            'last_name': last_name.title(),
-            'affiliation': affiliation,
-            'phone': phone,
-            'position': position
-        })
+    keys = []
+    for row_num, row in enumerate(reader):
+        if row_num == 0:
+            keys = [to_unicode(value).strip() for value in row]
+        else:
+            try:
+                regdata = {key: to_unicode(value).strip() for key,value in zip(keys,row)}
+            except ValueError:
+                raise UserValueError(_('Row {}: malformed CSV data - please check that the number of columns is correct')
+                                     .format(row_num))
+
+            if not regdata.get('email'):
+                raise UserValueError(_('Row {}: missing e-mail address').format(row_num))
+            regdata['email'] = regdata['email'].lower()
+            if not is_valid_mail(regdata['email'], multi=False):
+                raise UserValueError(_('Row {}: invalid e-mail address').format(row_num))
+            if not regdata.get('first_name') or not regdata.get('last_name'):
+                raise UserValueError(_('Row {}: missing first or last name').format(row_num))
+            if regdata['email'] in registered_emails:
+                raise UserValueError(_('Row {}: a registration with this email already exists').format(row_num))
+            if regdata['email'] in used_emails:
+                raise UserValueError(_('Row {}: email address is not unique').format(row_num))
+
+            used_emails.add(regdata['email'])
+            todo.append(regdata)
     return [create_registration(regform, data, notify_user=notify_users, skip_moderation=skip_moderation)
```

Comment:
Will the user get a meaningful error if they e.g. provide a field with the wrong name?

Reply:
Probably not  yet, error handling is not completed. But that's not different from the current situation if you have at least one mandatory extra field or removed  e.g. phone from your regform.